src/NServiceBus.Gateway/Routing/RouteToGatewayBehaviour.cs

```diff
@@ -1,23 +1,44 @@
 ﻿namespace NServiceBus
 {
     using System;
+    using System.Collections.Generic;
     using System.Threading.Tasks;
     using Gateway.Routing;
     using Pipeline;
     using Routing;
 
     class RouteToGatewayBehavior : Behavior<IRoutingContext>
     {
-        public RouteToGatewayBehavior(string gatewayAddress)
+        public RouteToGatewayBehavior(string gatewayAddress, IList<string> configuredSiteKeys)
         {
             this.gatewayAddress = gatewayAddress;
+            this.configuredSiteKeys = configuredSiteKeys;
         }
 
         public override Task Invoke(IRoutingContext context, Func<Task> next)
         {
-
             if (context.Extensions.TryGet(out RouteThroughGateway _))
             {
+                if (context.Message.Headers.TryGetValue(Headers.DestinationSites, out var siteKeyHeader))
+                {
+                    var siteKeys = siteKeyHeader.Split(',');
+
+                    var unknownSites = new List<string>();
```

Comment:
Can we change this so we're only allocating a collection if we actually have an unknown site? Otherwise we're always creating this and then just throwing it away for each message.